packages/web-component/src/components/read-along-component/read-along.tsx

```diff
@@ -1014,40 +1077,56 @@ export class ReadAlongComponent {
    *
    * A sentence element with one or more words
    */
-  Sentence = (props: { words: Node[], attributes: NamedNodeMap }): Element => {
-    if (!this.hasTextTranslations && props.attributes["class"]) {
-      this.hasTextTranslations = props.attributes["class"].value.match("translation") != null;
+  Sentence = (props: { sentenceData: Element }): Element => {
+    let words: ChildNode[] = Array.from(props.sentenceData.childNodes)
+    let sentenceID: string = props.sentenceData.id
+    if ((!this.hasTextTranslations) && props.sentenceData.hasAttribute('class')) {
+      this.hasTextTranslations = /translation/.test(props.sentenceData.getAttribute('class'));
     }
     let nodeProps = {};
-    if (props.attributes && props.attributes['xml:lang']) {
-
-      nodeProps['lang'] = props.attributes['xml:lang'].value
+    if (props.sentenceData.hasAttribute("xml:lang")) {
+      nodeProps['lang'] = props.sentenceData.getAttribute('xml:lang')
     }
-    if (props.attributes && props.attributes['lang']) {
-
-      nodeProps['lang'] = props.attributes['lang'].value
+    if (props.sentenceData.hasAttribute("lang")) {
+      nodeProps['lang'] = props.sentenceData.getAttribute('lang')
     }
 
     return <div {...nodeProps}
-      class={'sentence' + " " + (props.attributes["class"] ? props.attributes["class"].value : "")}>
+      class={'sentence' + " " + (props.sentenceData.hasAttribute('class') ? props.sentenceData.getAttribute('class') : "")}>
       {
         /* Here are the Word and NonWordText children */
-        props.words.map((child: Element, c) => {
+        words.map((child: Element, c) => {
 
           if (child.nodeName === '#text') {
             return <this.NonWordText text={child.textContent} attributes={child.attributes}
-              id={(props.attributes["id"] ? props.attributes["id"].value : "P") + 'text' + c} />
+              id={(props.sentenceData.hasAttribute('id') ? props.sentenceData.getAttribute('id') : "P") + 'text' + c} />
```

Comment:
Not a change request, but I wonder if this could be shortened with a short-circuit OR style construct, like

    id = {(props.sentenceData.getAttribute('id') || "P") ...

Caveat: I have no idea if that's even valid TS, or if it might throw an exception instead of doing what I expect, although https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute says `getAttribute()` returns `null` when the element does not exist, so it ought to work.

Reply:
Yes, I agree, it is nearly always better to use `getAttribute` than directly looking in `attributes`.  `undefined` is one of the many awful flaws of JavaScript and should be banished from the face of the earth

Reply:
Ah on further reflection, I disagree with using an or-construct, because what if `id` is `0` or `""` or something else un-truthy?  We should *always* explicitly test for things existing instead of relying on this, it is one of the worst flaws of JavaScript that you can access any attribute or key and just get `undefined` (at least Perl had the decency to not have different `undef` and `null` values)

Reply:
Right, that makes sense. So the code should stay as it was already, since it's indeed testing explicitly for what we actually want to know.